src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -596,17 +598,17 @@ class RichPairTraversableOnce[K, V](val t: TraversableOnce[(K, V)]) extends AnyV
   }
 }
 
-class RichPairIterator[K, V](val it: Iterator[(K, V)]) {
+class RichPairIterator[K : Ordering, V](val it: Iterator[(K, V)]) {
```

Comment:
#### Unifying

I'm not sure about tractability, it seems like in practice there are a handful of implicit parameters with the right "type". I glanced over [the docs on local type inference](http://scala-lang.org/files/archive/spec/2.11/06-expressions.html#local-type-inference), I don't see any discussion of implicit resolution, but there's this line under [_Implicits_](http://scala-lang.org/files/archive/spec/2.11/07-implicits.html#implicit-parameters):

> This discussion also shows that implicit parameters are inferred after any type arguments are inferred.

It seems the issue is also related to implicit conversion, because we have all three things happening at once here. In the current commit, the `PK` parameter to `localKeySort` is unconstrained, but the implicit conversion doesn't need to know the `PK`. This seems to be OK.

#### Class and Naming

Moving it to `RichOrderedKeyPairIterator` seems like the right approach. I can rename the `RichPairIterator` as well.